singularity-eos/base/table_bounds.hpp

```diff
@@ -0,0 +1,281 @@
+//------------------------------------------------------------------------------
+// © 2021-2024. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef SINGULARITY_EOS_BASE_TABLE_BOUNDS_HPP_
+#define SINGULARITY_EOS_BASE_TABLE_BOUNDS_HPP_
+#ifdef SINGULARITY_USE_SPINER
+
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include <ports-of-call/portability.hpp>
+#include <ports-of-call/portable_errors.hpp>
+#include <singularity-eos/base/fast-math/logs.hpp>
+#include <spiner/databox.hpp>
+#include <spiner/interpolation.hpp>
+
+namespace singularity {
```

Comment:
Would it make sense for this to be in its own namespace? My thinking is along a few lines:
* We do this in other utility functions such as the root finder.
* `Bounds` is a generic enough name that I'm hesitant for it to live naked in the `singularity` namespace
* These utilities only apply to tabular EOS, and specifically to those tabular EOS based on `spiner`, right?

Reply:
You could also rename the file `spiner_table_utils` and create a `spiner_utils` namespace.

Reply:
I am reluctant to name this `spiner_utils` as I don't think that's really what this is. I'll stick it in a `table_utils` namespace. How's that?

Reply:
They only apply to tabular EOS built off of spiner though, right?

Reply:
Currently that's true... But that would put this in `SpinerEOS` as it's a util for that, not for `Spiner` as a library. 

That said, I think this is more broadly useful for, e.g., computing log bounds and things like that.

Reply:
Also renamed file.